app/models/game.rb

```diff
@@ -28,6 +28,23 @@ def board_state
     board
   end
 
+  def in_check?(color)
+    #checking king
+    king = pieces.find_by(game_id: id, type:'King', color: color)
```

Comment:
don't need `game_id` here. Since you're in the game model, it already adds it automatically.